scwx-qt/source/scwx/qt/map/overlay_layer.cpp

```diff
@@ -173,27 +216,18 @@ void OverlayLayer::Initialize()
    }
 
    p->currentPosition_ = p->positionManager_->position();
-   auto coordinate     = p->currentPosition_.coordinate();
 
    // Geo Icons
-   p->geoIcons_->StartIconSheets();
-   p->geoIcons_->AddIconSheet(p->cursorIconName_);
-   p->geoIcons_->AddIconSheet(p->locationIconName_);
-   p->geoIcons_->FinishIconSheets();
-
-   p->geoIcons_->StartIcons();
-
-   p->cursorIcon_ = p->geoIcons_->AddIcon();
-   p->geoIcons_->SetIconTexture(p->cursorIcon_, p->cursorIconName_, 0);
-
-   p->locationIcon_ = p->geoIcons_->AddIcon();
-   p->geoIcons_->SetIconTexture(p->locationIcon_, p->locationIconName_, 0);
-   p->geoIcons_->SetIconAngle(p->locationIcon_,
-                              units::angle::degrees<double> {45.0});
-   p->geoIcons_->SetIconLocation(
-      p->locationIcon_, coordinate.latitude(), coordinate.longitude());
-
-   p->geoIcons_->FinishIcons();
+   auto& generalSettings = settings::GeneralSettings::Instance();
+   p->cursorScale_       = generalSettings.cursor_icon_scale().GetValue();
+   p->SetupGeoIcons();
+   generalSettings.cursor_icon_scale().RegisterValueChangedCallback(
```

Comment:
Needs a call to Unregister to go along with this. The callback will still be registered after the layer is deinitialized.

Alternatively, could use `generalSettings.cursor_icon_scale().changed_signal()`. Eventually, I'd like to replace all usages of the Register/Unregister style callbacks, as this removes the need to unregister anything (if a scoped connection is used). See alert_layer.cpp. I do plan on making a separate pull request to address all these.